.github/workflows/auto-cherry-pick.yml

```diff
@@ -0,0 +1,263 @@
+          echo "branch-name=${BRANCH_NAME}" >> $GITHUB_OUTPUT
+          echo "pr-number=${PR_NUMBER}" >> $GITHUB_OUTPUT
+
+          # Fetch and update branches
+          git fetch origin
+          git checkout develop
+          git pull origin develop
+          
+          if ! git checkout "${TARGET_BRANCH}"; then
+            echo "❌ Failed to checkout target branch: ${TARGET_BRANCH}"
+            exit 1
+          fi
+          git pull origin "${TARGET_BRANCH}"
+
+          # Create cherry-pick branch
+          git checkout -b "${BRANCH_NAME}"
+
+          # Attempt cherry-pick
+          echo "🍒 Cherry-picking ${SHORT_SHA} to ${TARGET_BRANCH}..."
+          
+          if git cherry-pick "${COMMIT_SHA}"; then
+            echo "✅ Cherry-pick successful"
+            git push origin "${BRANCH_NAME}"
+            echo "success=true" >> $GITHUB_OUTPUT
+            
+            # Create PR
+            PR_BODY="🍒 **Cherry-pick of #${PR_NUMBER}**
+
+          **Original PR**: #${PR_NUMBER}
+          **Original commit**: ${COMMIT_SHA}
+          **Target branch**: \`${TARGET_BRANCH}\`
+          **Cherry-pick branch**: \`${BRANCH_NAME}\`
+
+          ---
+
+          ${PR_TITLE}
+
+          > This cherry-pick was created automatically. Please review the changes and merge if appropriate.
+          > 
+          > If there are conflicts or issues, you may need to create a manual cherry-pick."
+
+            gh pr create \
+              --title "🍒 Cherry-pick: ${PR_TITLE} (${SHORT_SHA}) to ${TARGET_BRANCH}" \
+              --body "${PR_BODY}" \
+              --base "${TARGET_BRANCH}" \
+              --head "${BRANCH_NAME}" \
+              --label "cherry-pick" \
+              --label "automated"
+              
+            echo "✅ Cherry-pick PR created successfully"
+          else
+            echo "❌ Cherry-pick failed - conflicts detected"
+            echo "success=false" >> $GITHUB_OUTPUT
+            
+            # Clean up failed branch
+            git checkout "${TARGET_BRANCH}"
+            git branch -D "${BRANCH_NAME}" || true
+          fi
+
+      - name: Report failure
+        if: steps.cherry-pick.outputs.success == 'false'
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const targetBranch = '${{ matrix.target-branch }}';
+            const commitSha = '${{ steps.cherry-pick.outputs.commit-sha }}';
+            const shortSha = '${{ steps.cherry-pick.outputs.short-sha }}';
+            const prNumber = '${{ steps.cherry-pick.outputs.pr-number }}';
+            
+            // Comment on original PR about the failure
+            await github.rest.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: prNumber,
+              body: `❌ **Cherry-pick to \`${targetBranch}\` failed**
+
+            **Commit**: ${shortSha}
+            **Reason**: Merge conflicts detected
```

Comment:
Similar to the other comment, is the only failure mode a merge conflict? If not, let's make this more generic to avoid confusion.